src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -495,26 +501,65 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                     var elemRawType = vectorType.ItemType.RawType;
                     var srcNamedValueGetters = GetNamedOnnxValueGetters(input, _inputColIndices, _inputOnnxTypes, _inputTensorShapes);
                     if (vectorType.ItemType is TextDataViewType)
-                        return MakeStringTensorGetter(input, iinfo, srcNamedValueGetters, activeOutputColNames);
+                        return MakeStringTensorGetter(input, iinfo, srcNamedValueGetters, activeOutputColNames, outputCacher);
                     else
-                        return Utils.MarshalInvoke(MakeTensorGetter<int>, elemRawType, input, iinfo, srcNamedValueGetters, activeOutputColNames);
+                        return Utils.MarshalInvoke(MakeTensorGetter<int>, elemRawType, input, iinfo, srcNamedValueGetters, activeOutputColNames, outputCacher);
                 }
                 else
                 {
                     var type = _parent.Model.ModelInfo.OutputsInfo[_parent.MapDataViewColumnToOnnxOutputTensor(iinfo)].DataViewType.RawType;
                     var srcNamedValueGetters = GetNamedOnnxValueGetters(input, _inputColIndices, _inputOnnxTypes, _inputTensorShapes);
-                    return Utils.MarshalInvoke(MakeObjectGetter<int>, type, input, iinfo, srcNamedValueGetters, activeOutputColNames);
+                    return Utils.MarshalInvoke(MakeObjectGetter<int>, type, input, iinfo, srcNamedValueGetters, activeOutputColNames, outputCacher);
+                }
+            }
+
+            Delegate[] IRowMapper.CreateGetters(DataViewRow input, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Contracts.Assert(input.Schema == _inputSchema);
+
+                OnnxRuntimeOutputCacher outputCacher = new OnnxRuntimeOutputCacher();
+
+                int n = _outputColumns.Value.Length;
+                var result = new Delegate[n];
+                for (int i = 0; i < n; i++)
+                {
+                    if (!activeOutput(i))
+                        continue;
+                    result[i] = MakeGetter(input, i, activeOutput, outputCacher);
                 }
+                disposer = () =>
+                {
+                    outputCacher.Dispose();
+                };
+                return result;
             }
 
-            private class OnnxRuntimeOutputCacher
+            internal class OnnxRuntimeOutputCacher : IDisposable
```

Comment:
Just wondering, why did this needed to change from private to internal?

Reply:
Good catch. That is a left over from a previous experiment. I will change it back.

---
In reply to: [533789998](https://github.com/dotnet/machinelearning/pull/5518#discussion_r533789998) [](ancestors = 533789998)